### PR TITLE
Fix IPv6 usage in URLs

### PR DESCRIPTION
**Pull request by bentsi**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [ ] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/utils/common.py [hunk 1]
```diff
@@ -1483,3 +1483,10 @@ def download_encrypt_keys():
     for pem_file in ['CA.pem', 'SCYLLADB.pem', 'hytrust-kmip-cacert.pem', 'hytrust-kmip-scylla.pem']:
         if not os.path.exists('./data_dir/encrypt_conf/%s' % pem_file):
             ks.download_file(pem_file, './data_dir/encrypt_conf/%s' % pem_file)
+
+
+def normalize_ipv6_url(ip_address):
+    """adds square brackets on the IPv6 address in the URL"""
+    if ":" in ip_address:  # IPv6
```
Comment (fruch) [inline sdcm/utils/common.py]: maybe we could use 
https://docs.python.org/3/library/ipaddress.html#module-ipaddress
```python
    if ipaddress.ip_address(ip_address).version == 6:  # IPv6
        return "[%s]" % ip_address
    return ip_address
```

File: refresh_test.py [hunk 2]
```diff
@@ -76,12 +76,15 @@ def nodetool_refresh(self, node):
...
             node.run_nodetool("flush")
 
-    def check_timeout(self):
+    def check_timeout(self):  # pylint: disable=too-many-locals
         assert self.monitors.nodes, 'Monitor node should be set, we will try to get metrics from Prometheus server'
-        cmd = 'curl http://[%s]:9090/api/v1/query_range?query=scylla_storage_proxy_coordinator_read_timeouts&start=%s&end=%s&step=60s' % (
-            self.monitors.nodes[0].external_address, self.start, self.end)
+        base_url = "http://%s:9090/api/v1/query_range" % normalize_ipv6_url(self.monitors.nodes[0].external_address)
+        range_str = "&start={0.start}&end={0.end}".format(self)
+        cmd = ["curl",
```
Comment (fruch) [inline refresh_test.py]: why this test used curl ? and request ? seem like a very old test that wasn't run in a while...

**Comment by bentsi** [comment]
Provision test passed.

File: refresh_test.py (hunk 2 above)
Comment (bentsi) [reply refresh_test.py]: yes, I think we need to remove it. There is a refresh nemesis instead

File: sdcm/utils/common.py (hunk 1 above)
Comment (bentsi) [reply sdcm/utils/common.py]: We can, but in future when we will completely move to  Python 3

Comment (bentsi) [reply sdcm/utils/common.py]: this needs backport to branch-3.2

Comment (fruch) [reply sdcm/utils/common.py]: forgot the backports we need for this...

**Comment by bentsi** [comment]
@fruch updated the Hydra version. You can merge and backport

**Pull request merged**